app/src/reducers/vesselInfo.js

```diff
@@ -166,7 +171,7 @@ export default function (state = initialState, action) {
     }
     case SET_PINNED_VESSEL_TITLE: {
       const vesselIndex = state.vessels.findIndex(vessel => vessel.seriesgroup === action.payload.seriesgroup);
-      const newVessel = _.cloneDeep(state.vessels[vesselIndex]);
+      const newVessel = Object.assign({}, state.vessels[vesselIndex]);
```

Comment:
Object.assign makes a shallow copy of the object while cloneDeep makes a copy in memory of the whole objects (all nested objects). This caused typing a custom title to be very slow because it was copying the whole tracks array that is attached to the vessel.